File: frontend/src/api/agentActivity.ts
// Client API pour "Résumé Agent Room" et "Risque" (`GET
// /api/agents/decisions/recent`, `GET /api/risk/decisions/recent`, B26).
// Portée volontairement minimale — voir
// `backend/app/routers/agent_activity.py` : ce n'est pas l'Agent Room
// complet (Live Debate/Decision Details restent des placeholders honnêtes
// depuis B25, propriété de B28/B29), juste deux petits widgets de synthèse.

import { apiGet } from "./client";

export type AgentDecision = {
  id: string;
  strategy_id: string | null;
  agent_type: string;
  decision_type: string;
  outcome: string;
  confidence: number | null;
  created_at: string;
};

export type RecentAgentDecisionsResponse = {
  decisions: AgentDecision[];
};

export type RiskDecision = {
  id: string;
  agent_decision_id: string;
  outcome: string;
  reasons: unknown[];
  created_at: string;
};

export type RecentRiskDecisionsResponse = {
  decisions: RiskDecision[];
};

export async function fetchRecentAgentDecisions(limit = 5): Promise<RecentAgentDecisionsResponse> {
  return apiGet<RecentAgentDecisionsResponse>(`/api/agents/decisions/recent?limit=${limit}`);
}

export async function fetchRecentRiskDecisions(limit = 5): Promise<RecentRiskDecisionsResponse> {
  return apiGet<RecentRiskDecisionsResponse>(`/api/risk/decisions/recent?limit=${limit}`);
}

File: backend/app/schemas/agent_activity.py
"""Schémas des résumés d'activité "agents" et "risque" (§B26 "Résumé Agent
Room" et "Risque").

Même portée volontairement minimale que `schemas/orders.py` : ce n'est PAS
le futur écran Agent Room complet (Live Debate, Ask Ziko AI, Decision
Details restent des placeholders honnêtes depuis B25, propriété de B28/B29)
mais juste de quoi peupler deux petits widgets de synthèse sur le tableau
de bord principal, à partir de données déjà réellement écrites par le
Strategy Agent/Risk Critic Agent (B13/B14, table `agent_decisions`) et le
Risk Engine déterministe (B15, table `risk_decisions`)."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel


class AgentDecisionOut(BaseModel):
    id: uuid.UUID
    strategy_id: uuid.UUID | None
    agent_type: str
    decision_type: str
    outcome: str
    confidence: int | None
    created_at: datetime

    model_config = {"from_attributes": True}


class RecentAgentDecisionsResponse(BaseModel):
    decisions: list[AgentDecisionOut]


class RiskDecisionOut(BaseModel):
    id: uuid.UUID
    agent_decision_id: uuid.UUID
    outcome: str
    reasons: list
    created_at: datetime

    model_config = {"from_attributes": True}


class RecentRiskDecisionsResponse(BaseModel):
    decisions: list[RiskDecisionOut]
